File: MasGlobal.Assessment.Entities/Employee.cs
using MasGlobal.Assessment.Entities.Interfaces;
using System;

namespace MasGlobal.Assessment.Entities
{

    public class Employee : IContracts
    {
        public Employee()
        {
        }

        public Employee(Employees.EmployeeResponse item)
        {
            id = item.id;
            name = item.name;
            contractTypeName = item.contractTypeName;
            roleId = item.roleId;
            roleName = item.roleName;
            roleDescription = item.roleDescription;
            hourlySalary = item.hourlySalary;
            monthlySalary = item.monthlySalary;
        }

        public int id { get; set; }

        public string name { get; set; }

        public string contractTypeName { get; set; }

        public int roleId { get; set; }

        public string roleName { get; set; }

        public string roleDescription { get; set; }

        public long hourlySalary { get; set; }

        public long monthlySalary { get; set; }

        public long anualSalary { get; set; }
    }
}

File: MasGlobal.Assessment.UI/Scripts/JS/Employee.js
$(document).ready(function () {
    $("#searchEmployee").click(function () {
        ShowModal();
        let employeeId = $("#idEmployee").val();
        GetEmployees(employeeId);
    });
});

async function GetEmployees(employeeId) {
    //$("#tbodyEmployees").html('<tr><td colspan = "5" class= "text-center" > No Records</td ></tr>');
    $("#tbodyEmployees").html('');
    try {
        await fetch('/api/employee/' + employeeId)
            .then(function (responseData) {
                console.log(responseData);
                return responseData.json();
            }
            ).then(function (response) {
                if (response.status === true) {
                    DrawRecord(response.employees);
                } else {
                    swal("Error", response.message, "error");
                }
                HideModal();
                
            }
            ).catch(function (error) {
                swal("Error", "Something is wrong", "error");
                HideModal();
            });

    } catch (err) {
        alert('falle');
        HideModal();
    }

}

function DrawRecord(records) {
    if (records !== undefined) {
        if (records.length > 0) {
            records.map(function (item) {
                let rowText = `<tr><td>${item.id}</td><td>${item.name}</td><td>${item.contractTypeName}</td><td>${item.roleName}</td><td>${item.anualSalary}</td></tr>`;
                $("#tbodyEmployees").append(rowText);
            });
            HideModal();
        }
    }
}

function ShowModal() {
    $("#modalLoader").removeClass("withOutModal").addClass("withModal");
};

function HideModal() {
    $("#modalLoader").removeClass("withModal").addClass("withOutModal");

};